src/VisualStudio/CSharp/Test/ProjectSystemShim/CPS/SourceFileHandlingTests.cs

```diff
@@ -57,5 +58,350 @@ public void AddRemoveAdditionalFile_CPS()
                 Assert.Empty(GetCurrentAdditionalDocuments());
             }
         }
+
+        [WpfFact]
+        [Trait(Traits.Feature, Traits.Features.ProjectSystemShims)]
+        public void ReorderSourceFiles_CPS()
+        {
+            using (var environment = new TestEnvironment())
+            using (var project = CreateCSharpCPSProject(environment, "project1"))
+            {
+                IEnumerable<Document> GetCurrentDocuments() => environment.Workspace.CurrentSolution.Projects.Single().Documents;
+
+                Assert.Empty(GetCurrentDocuments());
+
+                var sourceFileFullPath1 = @"c:\source1.cs";
+                var sourceFileFullPath2 = @"c:\source2.cs";
+                var sourceFileFullPath3 = @"c:\source3.cs";
+                var sourceFileFullPath4 = @"c:\source4.cs";
+                var sourceFileFullPath5 = @"c:\source5.cs";
+                project.AddSourceFile(sourceFileFullPath1);
+                project.AddSourceFile(sourceFileFullPath2);
+                project.AddSourceFile(sourceFileFullPath3);
+                project.AddSourceFile(sourceFileFullPath4);
+                project.AddSourceFile(sourceFileFullPath5);
+
+                project.RemoveSourceFile(sourceFileFullPath2);
+                project.AddSourceFile(sourceFileFullPath2);
+
+                project.RemoveSourceFile(sourceFileFullPath4);
+                project.AddSourceFile(sourceFileFullPath4);
```

Comment:
You may want to add an assertion that the order _isn't_ the sorted order, so that way we know that ReorderSourceFiles actually did something.

Reply:
Makes sense.